<a href="https://colab.research.google.com/github/ElofssonLab/kb8029-book/blob/main/notebooks/session02-discussion-2.ipynb" style="display:inline-block;padding:10px 18px;background-color:#F9AB00;color:#000000;font-weight:bold;text-decoration:none;border-radius:6px;font-family:sans-serif;font-size:14px;">&#9654;&nbsp; Open in Google Colab</a>

# Session 2 — In-class discussion problem (2 of 3)

Work this out **by hand in your group first** — then run the code cell
to check your answer before presenting.

## Is a strong correlation from 5 data points trustworthy?

A group measures (standardized) length and GC content for just 5
genes:

```
x = [1.83, -3.08, 0.96, 0.07, 1.32]
y = [1.14, -0.44, 0.50, -0.27, 1.01]
```

`scipy.stats.pearsonr` gives r ≈ 0.85 — a strong correlation.

**As a group:**

1. Would you call r = 0.85 convincing evidence of a real relationship?
2. Check the p-value. Is it below the conventional 0.05 threshold?
3. Predict: if 50 *independent* genes were measured, with the same
   underlying correlation, would the p-value change? If so, in which
   direction, and why?

Run the cell below to check your answers.

In [1]:
import numpy as np
from scipy import stats

x5 = np.array([1.83, -3.08, 0.96, 0.07, 1.32])
y5 = np.array([1.14, -0.44, 0.50, -0.27, 1.01])
r5, p5 = stats.pearsonr(x5, y5)
print(f"n = 5:   r = {r5:.2f}, p = {p5:.3f}")

# What the p-value means (Session 1): how often do 5 points with NO relationship
# give a correlation at least this strong, just by chance?
rng = np.random.default_rng(1)
null_r = np.array([stats.pearsonr(*rng.normal(size=(2, 5)))[0] for _ in range(20000)])
print(f"         5 unrelated points give |r| >= {r5:.2f} in {np.mean(np.abs(null_r) >= r5):.1%} of 20,000 tries")

# 50 independent genes from a population with the same correlation (0.85)
x50, y50 = rng.multivariate_normal([0, 0], [[1, 0.85], [0.85, 1]], size=50).T
r50, p50 = stats.pearsonr(x50, y50)
print(f"n = 50:  r = {r50:.2f}, p = {p50:.0e}")

n = 5:   r = 0.85, p = 0.066
         5 unrelated points give |r| >= 0.85 in 6.8% of 20,000 tries
n = 50:  r = 0.88, p = 9e-17


**Discussion point:** with only 5 points, r ≈ 0.85 but p ≈ 0.066 (about
0.07): *not* significant at the conventional 0.05 threshold, despite
looking like a strong correlation. The simulation shows what that
p-value means: five points with no relationship at all give a
correlation this strong about 7% of the time. With 50 independent genes
and the same underlying correlation, r stays about as strong (0.88 in this sample) but
p drops to about $10^{-16}$. The *strength* of the correlation hardly
changed; what changed is how sure you can be that it isn't chance.

(Measuring the same 5 genes ten times over would not help: that is 5
genes, not 50. The points must be independent.)

**You will meet this again.** In the Session 2 mini challenge, six globin
mRNAs give r = 0.78 but p = 0.07; in the lab, twelve globins give a
similar r (0.79) with p = 0.002.